### PR TITLE
leveldb tinylicious tests

### PR DESCRIPTION
**Pull request by kurtb**
Adds leveldb tests for tinylicious. Does so by expanding on the previous memory collection tests. By making use of the DbFactory we can target both database types with the same test code. A simple test factory is needed for custom init and cleanup (i.e. deleting the leveldb directory).

### CODE REVIEW

File: server/tinylicious/src/test/dbFactory.spec.ts [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+            },
+            {
+                name: "levelDb",
+                create: () => {
+                    const levelDir = fs.mkdtempSync(path.join(os.tmpdir(), "level-"));
+
+                    return {
+                        value: {
+                            db: {
+                                inMemory: true,
```
Comment (tanviraumi) [inline server/tinylicious/src/test/dbFactory.spec.ts:72]: false?

**Review by tanviraumi — COMMENTED** [review]
Looks good but the in memory config for level db was set to true so the tests are probably only running in memory.

File: server/tinylicious/src/test/dbFactory.spec.ts (hunk 1 above)
Comment (kurtb) [reply server/tinylicious/src/test/dbFactory.spec.ts]: Ah - good catch! Fixed things up to actually turn leveldb on...

As part of this also needed to tweak the tests a bit. MongoDB is the basis for these DB APIs. And both in memory/level are subsets of the Mongo API. Tinylicious/r11s obviously don't use the full set of functionality.

The test now better captures the usage of each collection and serves as a bit of a conformance test should anyone add another use case.

Having a more constrained DB API may be nice to better highlight the actual required functionality.

**Review by tanviraumi — APPROVED** [review]
Thank you for the refactoring and tests. The tests are more comprehensive than our current usage of levelDb which is great.

**Comment by tylerbutler** [comment]
Our build pipeline is having some issues unrelated to this PR. Assigned this bug to myself to merge once our build issues are resolved.

**Comment by tylerbutler** [comment]
Closing and reopening to trigger PR validation.

**Pull request merged**